dags/sentinel-2_data_transfer.py

```diff
@@ -38,57 +43,167 @@ def extract_src_key(src_url):
+    content = content_object.get()['Body'].read().decode('utf-8')
+    data = json.loads(content)
+    return data
+
+def africa_extent():
+    """
+    Load Africa boundaries
+    :return: Shape object of the boundaries
+    """
+
+    boundaries = json.load(open("data/africa-extent.json"))
+    return shape(boundaries["features"][0]["geometry"])
+
+def scene_stac_utm(stac):
+    """
+    Extract the UTM zone from the STAC file
+    :return: UTM zone
+    """
+
+    return stac['properties']['proj:epsg']
+
+def reproject_scene_geom_to_epsg_4326(stac, scene_geometry):
+    """
+    Reproject CRS of the scene to WGS84 (same as the Africa extent)
+    :param stac: STAC file
+    :param scene_geometry: Scene geometry
+    :return: Scene geometry reprojected to epsg:4326
+    """
+
+    src_epsg = scene_stac_utm(stac)
+    src_gcs = "epsg:" + str(src_epsg)
+
+    e_target = CRS('epsg:4326')
+    e_source = CRS(src_gcs)
+    project = Transformer.from_proj(Proj(e_source), Proj(e_target))
+    return transform(project.transform, scene_geometry) 
+
+def is_in_africa(s3_hook, scene_geometry, africa_footprint, stac_file):
+    """
+    Check whether a scene falls within Africa boundary
+    :param s3_hook: hook to interact with S3
+    :param scene_geometry: Shape object of the scene Geometry
+    :param africa_footprint: A dictionary containing Africa boundary
+    :param stac_file: path to the STAC file on S3
+    :return: a boolean indicating whether a scene falls within Africa boundary
+    """
+
+    stac = load_stac(s3_hook, stac_file)
+    src_epsg = scene_stac_utm(stac)
+    scene_epsg_4326 = reproject_scene_geom_to_epsg_4326(stac, scene_geometry)
+    # Exclude scenes on the anti-meridian
+    merdian_180 = src_epsg not in list(default_args["crs_black_list"])
+    return  scene_geometry.intersects(africa_footprint) & merdian_180
+
 def copy_s3_objects(ti, **kwargs):
     """
     Copy objects from a s3 bucket to another s3 bucket.
     :param ti: Task instance
     """
-  
+
     s3_hook = S3Hook(aws_conn_id=dag.default_args['aws_conn_id'])
-    messages = ti.xcom_pull(key='messages', task_ids='sqs_sensor')
-    if messages:
-        for rec in messages['Messages']:
-            body = json.loads(rec['Body'])
-            message = json.loads(body['Message'])
-
-            # Extract URL of the json file        
-            urls = [message["links"][0]["href"]]
+    messages = ti.xcom_pull(key='Messages', task_ids='test_trigger_dagrun')
+    # Load Africa footprint
+    africa_footprint = africa_extent()
+
+    for rec in messages:
+        body = json.loads(rec)
+        message = json.loads(body['Message'])
+        scene_geometry = shape(message["geometry"])
+
+        # Extract URL of the json file
+        urls = [message["links"][0]["href"]]
+        is_in_africa_flag = is_in_africa(s3_hook, scene_geometry,
+                                                    africa_footprint, urls[0])
```

Comment:
This indentation is a bit ugly... but not a big deal!

Reply:
Oh, it is ugly.  Will fix and push later.